platforms/documentation/docs/src/snippets/dependencyManagement/workingWithDependencies-iterateDependencies/groovy/build.gradle

```diff
@@ -13,10 +13,16 @@ dependencies {
 
 // tag::iteration-task[]
 tasks.register('iterateDeclaredDependencies') {
+    def dependencySet = configurations.scm.dependencies
+    def artifactInfo = dependencySet.collect {
+        [
+            group: it.group,
+            name: it.name,
+            version: it.version
+        ]
+    }
```

Comment:
🤔  If this is part of some userguide page, I'd like us to find a better solution, if possible.
Is `dependencySet` to something supported by CC?

Reply:
In other sample, we seem to be able bring the whole `scm` inside the `doLast` block

Reply:
>If this is part of some userguide page

It isn't, maybe we should just convert these to normal integration tests.

Reply:
> In other sample, we seem to be able bring the whole scm inside the doLast block

In Groovy only, though. I could make the Groovy sample more similar to the Kotlin version though (and [extract artifact info at config time](https://github.com/gradle/gradle/pull/29849/files#diff-18a2c1132aaf69cf6990cce4b9af30e332c83414471c637f68735b9668639dedR17)).

Reply:
I think not introducing an ad-hoc map or a `Triple` would be preferable, but if it's not user-facing, then it does not matter that much.

I agree with Mike, then maybe it should be an integration test instead.

Reply:
>  If this is part of some userguide page, I'd like us to find a better solution, if possible.

@alllex What is your concern here? Projecting the required data from unsupported Gradle objects as POJOs is a common pattern in making builds CC-compatible, right?

> maybe we should just convert these to normal integration tests.

@mlopatkin Sorry, I don't follow this either. Could you please clarify?

Reply:
> I think not introducing an ad-hoc map or a Triple would be preferable, but if it's not user-facing,

@alllex Only saw this now, sorry. Still unclear why it is undesirable though.

Reply:
> Sorry, I don't follow this either. Could you please clarify?

AFAIR, these `workingWithDependencies` tests were part of the software model documentation that is no longer available. They still have a value as integration tests, though, because we never truly removed the SM code. But having them in samples may confuse readers who don't know all the backstory, so we can just replace these sample-based tests with a usual integration test. I have no idea why we didn't do this when we were fixing CC compatibility of samples :(

Reply:
Ok, thanks, will open a chore issue for that

Reply:
Opened https://github.com/gradle/gradle/issues/29864